src/interface/src/styleguide/paginator/paginator.component.ts

```diff
@@ -67,13 +75,25 @@ export class PaginatorComponent implements OnInit {
     this.calcButtonLabels();
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['pageCount']) {
+      this.calcButtonLabels();
+    }
+  }
+
   getTotalPages(): number {
     return this.pageCount;
   }
 
   calcButtonLabels(): void {
     const curPages = this.pageCount;
     const buttonsToShow = Math.min(curPages, this.defaultButtonsToShow);
+
+    // in case the results-per-page changes, ensure that the
+    // currently selected page number doesn't exceed
+    // the number of pages available
+    this.selectedPage = Math.min(curPages, this.selectedPage);
```

Comment:
This wasn't part of the initial reported issue, but I thought we should add this.